mex/common/types/text.py

```diff
@@ -30,7 +54,7 @@ class Text(BaseModel):
     """
 
     value: Annotated[str, Field(min_length=1)]
-    language: TextLanguage | None = None
+    language: RestrictedTextLanguage | TextLanguage | None = None
```

Comment:
changing the type here would have quite deep implications. could you instead convert RestrictedTextLanguage back to an instance of TextLanguage after the detection?

```suggestion
    language: TextLanguage | None = None
```